front/pages/w/[wId]/builder/assistants/dust.tsx

```diff
@@ -0,0 +1,284 @@
+      sendNotification({
+        title: "Dust Assistant",
+        description: `@${agent.name} is only available on our paid plans. Contact us at team@dust.tt to get access.`,
+        type: "error",
+      });
+      return;
+    }
+    if (agent.status === "disabled_missing_datasource") {
+      sendNotification({
+        title: "Dust Assistant",
+        description: `@${agent.name} is not available because you have not configured any data sources.`,
+        type: "error",
+      });
+      return;
+    }
+    const res = await fetch(
+      `/api/w/${owner.sId}/assistant/global_agents/${agent.sId}`,
+      {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          status:
+            agent.status === "disabled_by_admin"
+              ? "active"
+              : "disabled_by_admin",
+        }),
+      }
+    );
+
+    if (!res.ok) {
+      const data = await res.json();
+      window.alert(`Error toggling Assistant: ${data.error.message}`);
+      return;
+    }
+
+    await mutateAgentConfigurations();
+  };
+
+  const updateDatasourceSettings = async (
+    settings: {
+      assistantDefaultSelected: boolean;
+    },
+    dataSource: DataSourceType
+  ) => {
+    const res = await fetch(
+      `/api/w/${owner.sId}/data_sources/${dataSource.name}`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(settings),
+      }
+    );
+    if (!res.ok) {
+      const err = (await res.json()) as { error: APIError };
+      window.alert(
+        `Failed to update the Data Source (contact team@dust.tt for assistance) (internal error: type=${err.error.type} message=${err.error.message})`
+      );
+    }
+    await mutateDataSources();
+    await mutateAgentConfigurations();
+  };
+
+  return (
+    <AppLayout
+      hideSidebar
+      user={user}
+      owner={owner}
+      gaTrackingId={gaTrackingId}
+      topNavigationCurrent="settings"
+      subNavigation={subNavigationAdmin({ owner, current: "assistants" })}
+      titleChildren={
+        <AppLayoutSimpleCloseTitle
+          title="Manage Dust Assistant"
+          onClose={async () => {
+            await router.push(`/w/${owner.sId}/builder/assistants`);
+          }}
+        />
+      }
+    >
+      <div className="pt-8" />
+      <PageHeader
+        title="Dust Assistant"
+        icon={LogoSquareColorLogo}
+        description="The Dust assistant is a general purpose assistant that has context on your company data."
+      />
+      <div className="flex flex-col space-y-8 pb-8 pt-8">
+        <div className="flex w-full flex-col gap-4">
+          {!!dataSources.length && (
+            <>
+              <SectionHeader
+                title="Availability"
+                description="The Dust assistant requires data sources or connections to operate."
```

Comment:
The Dust assistant requires at least one data source or connection to be enabled.